File: library/lib_module_common/src/main/java/org/cchao/common/utils/LanguageUtils.java
package org.cchao.common.utils;

import android.content.res.Configuration;

import org.cchao.common.IApplication;

import java.util.Locale;

public class LanguageUtils {

    private LanguageUtils() {
    }

    public static String getStringByLocal(int resId, String localStr) {
        return getStringByLocal(resId, new Locale(localStr));
    }

    public static String getStringByLocal(int resId, Locale locale) {
        Configuration configuration = new Configuration(IApplication.getInstance().getResources().getConfiguration());
        configuration.setLocale(locale);
        return IApplication.getInstance().createConfigurationContext(configuration).getResources().getString(resId);
    }
}

File: library/lib_module_common/src/main/java/org/cchao/common/utils/LanguageUtils.kt
package org.cchao.common.utils

import android.content.res.Configuration

import org.cchao.common.IApplication

import java.util.Locale

object LanguageUtils {

    fun getStringByLocal(resId: Int, localStr: String): String {
        return getStringByLocal(resId, Locale(localStr))
    }

    fun getStringByLocal(resId: Int, locale: Locale): String {
        val configuration = Configuration(IApplication.instance.resources.configuration)
        configuration.setLocale(locale)
        return IApplication.instance.createConfigurationContext(configuration).resources.getString(resId)
    }
}
